nautobot/extras/models/jobs.py

```diff
@@ -653,16 +652,19 @@ def enqueue_job(
         if celery_kwargs is not None:
             job_celery_kwargs.update(celery_kwargs)
 
+        # When running synchronously (explicitly on this call or globally), the celery worker
+        # setup_nautobot_job_logging() doesn't get called by a worker signal, so we need to do it here.
         if synchronous or getattr(settings, "CELERY_TASK_ALWAYS_EAGER", False):
+            setup_nautobot_job_logging(None, None, app.conf)
+
+        if synchronous:
             # synchronous tasks are run before the JobResult is saved, so any fields required by
             # the job must be added before calling `apply()`
             job_result.celery_kwargs = job_celery_kwargs
             job_result.save()
 
             # setup synchronous task logging
             redirect_logger = get_logger("celery.redirected")
```

Comment:
This comment can be removed and the following line moved down below the next comment

Reply:
Good thought. Will do once CI verifies that the fix is otherwise valid.